Neos.Neos/Classes/PendingChangesProjection/ChangeProjection.php

```diff
@@ -218,18 +218,17 @@ private function whenNodeAggregateWasMoved(NodeAggregateWasMoved $event): void
         }
 
         $affectedDimensionSpacePoints = iterator_to_array($event->succeedingSiblingsForCoverage->toDimensionSpacePointSet());
-        $arbitraryDimensionSpacePoint = reset($affectedDimensionSpacePoints);
-        if ($arbitraryDimensionSpacePoint instanceof DimensionSpacePoint) {
+        foreach ($affectedDimensionSpacePoints as $affectedDimensionSpacePoint) {
             // always the case due to constraint enforcement (at least one DSP is selected and must have a succeeding sibling or null)
 
-            // WORKAROUND: we simply use the event's first DSP here as the origin dimension space point.
+            // WORKAROUND: we simply use the events DSPs here as the origin dimension space point.
             // But this DSP is not necessarily occupied.
             // @todo properly handle this by storing the necessary information in the projection
 
```

Comment:
This statement is true,

> But this DSP is not necessarily occupied.

but its not problematic as well publish all dimensions either way. The dimension information will only be used as indicated for the Ui.

https://github.com/neos/neos-development-collection/pull/5385

```suggestion
            // We simply use the events DSPs here to store them as `Change` in even if the DSP is not necessarily  occupied.
            // this is not problematic as the DSP should only be used for providing additional information where a change has effects instead of locating its origin
            // todo possibly rename in the `Change` the field to '$affectedDimensionSpacePoint' field instead, as well use it now like that.

```